luaui/Widgets/cmd_blueprint.lua

```diff
@@ -864,6 +864,45 @@ local function handleSpacingAction(_, _, args)
 	end
 end
 
+local function handleBlueprintSelectAction(cmd, optLine, optWords, data, isRepeat, release, actions)
+	local targetIndex = tonumber(optLine)
+
+	local function selectIndex(index)
+		setSelectedBlueprintIndex(index)
+		Spring.SetActiveCommand(
+			Spring.GetCmdDescIndex(CMD_BLUEPRINT_PLACE),
+			1,
+			true,
+			false,
+			false,
+			false,
+			false,
+			false
+		)
+	end
+
+	if targetIndex ~= nil then
+		if targetIndex <= #blueprints then
+			selectIndex(targetIndex)
+			return true
+		else
+			return false
+		end
+	end
+
+	local targetName = optLine
+	if targetName ~= nil then
+		for blueprintIndex, blueprint in ipairs(blueprints) do
+			if blueprint.name == targetName then
+				selectIndex(blueprintIndex)
+				return true
+			end
+		end
```

Comment:
Wouldn't it make more sense to compile a map of blueprint name -> blueprint ID, so they can be selected directly by name?

Reply:
That would probably be a bit faster, it would just need a bit more code, and performance shouldn't be an issue here. I'm ambivalent which way it's done, so I don't mind changing it.